src/Serialization/Address.js

```diff
@@ -0,0 +1,148 @@
+    return maybe.just(ret);
+};
+
+exports.rewardAddressFromBytesImpl = maybe => bytes => {
+    const ret = null;
+    try {
+        const addr = CardanoWasm.Address.from_bytes(bytes);
+        ret = CardanoWasm.RewardAddress.from_address(addr);
+    }
+    catch (e) {
+        console.log(e);
+    }
+    if (ret == null) {
+        return maybe.nothing;
+    }
+    return maybe.just(ret);
+};
+
+exports.baseAddressFromBech32Impl = maybe => str => {
+    const ret = null;
+    try {
+        const addr = CardanoWasm.Address.from_bech32(str);
+        ret = CardanoWasm.BaseAddress.from_address(addr);
+    }
+    catch (e) {
+        console.log(e);
+    }
+    if (ret == null) {
+        return maybe.nothing;
+    }
+    return maybe.just(ret);
+};
+
+exports.rewardAddressFromBech32Impl = maybe => str => {
+    const ret = null;
+    try {
+        const addr = CardanoWasm.Address.from_bech32(str);
+        ret = CardanoWasm.RewardAddress.from_address(addr);
+    }
+    catch (e) {
+        console.log(e);
+    }
+    if (ret == null) {
+        return maybe.nothing;
+    }
+    return maybe.just(ret);
+};
+
+exports.headerCheckBaseAddr = maybe => checks => bytes => baseAddr => {
+    // NOTE from CIP-19 and CSL codebase:
+    // shelley payment addresses:
+    // bit 7: 0
+    // bit 6: base/other
+    // bit 5: pointer/enterprise [for base: stake cred is keyhash/scripthash]
+    // bit 4: payment cred is keyhash/scripthash
+    // bits 3-0: network id
+    //
+    // reward addresses:
+    // bits 7-5: 111
+    // bit 4: credential is keyhash/scripthash
+    // bits 3-0: network id
+    //
+    // byron addresses:
+    // bits 7-4: 1000
+    const paymcheck = maybe.from(null, checks.payment);
+    if (paymcheck != null && !checkBitSet(bytes, 4, paymcheck)){
+        return maybe.nothing;
+    }
+    const delegcheck = maybe.from(null, checks.delegation);
+    if (delegcheck != null && !checkBitSet(bytes, 3, delegcheck)) {
+        return maybe.nothing;
+    }
+    return maybe.just(baseAddr);
+};
+
+exports.headerCheckRewardAddr = maybe => checks => bytes => baseAddr => {
+    // NOTE from CIP-19 and CSL codebase:
+    // shelley payment addresses:
+    // bit 7: 0
+    // bit 6: base/other
+    // bit 5: pointer/enterprise [for base: stake cred is keyhash/scripthash]
+    // bit 4: payment cred is keyhash/scripthash
+    // bits 3-0: network id
+    //
+    // reward addresses:
+    // bits 7-5: 111
+    // bit 4: credential is keyhash/scripthash
+    // bits 3-0: network id
+    //
+    // byron addresses:
+    // bits 7-4: 1000
+    const paymcheck = maybe.from(null, checks.payment);
+    if (paymcheck != null && !checkBitSet(bytes, 4, paymcheck)) {
+        return maybe.nothing;
+    }
+    return maybe.just(baseAddr);
+};
+
+
+exports.toAddressCslUnsafe = addresType => {
+    return addresType.to_address();
+};
+
+exports.addressBytesImpl = baseAddr => {
+    return baseAddr.to_bytes();
+};
+
+exports.addressBech32Impl = baseAddr => {
```

Comment:
Let's stick to `_` at the end of FFI implementations, as suggested [here](https://github.com/Plutonomicon/cardano-browser-tx/pull/51#discussion_r791712120)

Reply:
ok

Reply:
Sorry, I messed up this suggestion. The original comment by @vvtran was refering to `src/Ogmious.js`

> Minor nitpick, the rest of the codebase prepends FFIs with an underscore, we should probably stick to one convention.

Reply:
^^

Reply:
Reverted to the original for now.

Reply:
applied prefix for functions not intented for export